### PR TITLE
Enforce repo management permission

### PR DESCRIPTION
**Pull request by gmazoyer**

### CODE REVIEW

File: backend/infrahub/permissions/report.py [hunk 1]
```diff
@@ -17,26 +17,51 @@
...
+                else BranchRelativePermissionDecision.DENY
+            )
+        if node.kind in (InfrahubKind.GENERICREPOSITORY, InfrahubKind.REPOSITORY, InfrahubKind.READONLYREPOSITORY) or (
+            node.inherit_from and InfrahubKind.GENERICREPOSITORY in node.inherit_from
+        ):
+            return (
+                BranchRelativePermissionDecision.ALLOW
+                if global_permission_report[GlobalPermissions.MANAGE_REPOSITORIES]
+                else BranchRelativePermissionDecision.DENY
+            )
```
Comment (gmazoyer) [inline backend/infrahub/permissions/report.py]: Probably not the prettiest code to achieve that.

**Comment by lykinsbd** [comment]
Looks like the E2E tests don't like something about this, they've failed with this error twice: 

```
  1 failed
    [chromium] › permissions/role-management.spec.ts:7:3 › Users & Permissions - Admin › should be allowed to add accounts 
```

File: backend/infrahub/permissions/report.py [hunk 2]
```diff
@@ -17,26 +18,51 @@
...
         return BranchRelativePermissionDecision.ALLOW
 
     decision = backend.report_object_permission(
         permissions=permissions["object_permissions"], namespace=node.namespace, name=node.name, action=action
     )
 
     # What do we do if edit default branch global permission is set?
-    # if can_edit_default_branch:
+    # if global_permission_report[GlobalPermissions.EDIT_DEFAULT_BRANCH]:
     #     decision |= PermissionDecisionFlag.ALLOW_DEFAULT
```
Comment (ajtmccarty) [inline backend/infrahub/permissions/report.py]: can this comment be removed? should it be a Jira issue?

**Review by ajtmccarty — APPROVED** [review]

File: backend/infrahub/permissions/report.py (hunk 2 above)
Comment (gmazoyer) [reply backend/infrahub/permissions/report.py]: You're right, logged IFC-934 about it.

**Pull request merged**